lua/barbar/render.lua

```diff
@@ -139,6 +139,28 @@ local function hl_tabline(group)
   return '%#' .. group .. '#'
 end
 
+-- Digits for optional styling of buffer_number and buffer_index.
+local SUPERSCRIPT_DIGITS = { '⁰', '¹', '²', '³', '⁴', '⁵', '⁶', '⁷', '⁸', '⁹' }
+local SUBSCRIPT_DIGITS = { '₀', '₁', '₂', '₃', '₄', '₅', '₆', '₇', '₈', '₉' }
+
+---@param num number
+---@param style barbar.config.options.icons.numbers.value
+local function style_number(num, style)
+  if style ~= 'superscript' and style ~= 'subscript' then
+    return tostring(num)
+  end
+
+  local styled_number = ''
+
+  for digit in string.gmatch(tostring(num), "%d") do
+    digit = digit + 1
+    styled_number = styled_number .. (style == "superscript" and SUPERSCRIPT_DIGITS[digit]
+      or style == "subscript" and SUBSCRIPT_DIGITS[digit])
+  end
```

Comment:
lint: Mixed double-quotes, we use single quotes.

Reply:
Oh sorry about that, I had to turn off my formatter. I'll fix that quickly!

Is there a configuration for auto formatting that the repo is following? I didn't found guidelines or settings.

Reply:
We don't have an .editorconfig, but seeing as Nvim 0.9 is adding support, we should probably do that.